presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java

```diff
@@ -1341,6 +1374,24 @@ private static OGCGeometry geomFromBinary(Slice input)
         return geometry;
     }
 
+    private static ByteBuffer getShapeByteBuffer(Slice input)
+    {
+        int offset = HADOOP_SHAPE_SIZE_WKID + HADOOP_SHAPE_SIZE_TYPE;
+        if (input.length() <= offset) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Hadoop shape input is too short");
+        }
+        return input.toByteBuffer(offset, input.length() - offset).slice().order(ByteOrder.LITTLE_ENDIAN);
+    }
+
+    private static int getWktExportFlags(Slice input)
+    {
+        byte hadoopShapeType = input.getByte(HADOOP_SHAPE_SIZE_WKID);
+        if (hadoopShapeType < 0 || hadoopShapeType >= HADOOP_SHAPE_TYPES.length) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid hadoop shape type");
```

Comment:
```suggestion
            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid Hadoop shape type: " + hadoopShapeType);
```
Also, related test error message.